src/Mvc/Mvc.Core/src/Infrastructure/ObjectResultExecutor.cs

```diff
@@ -2,14 +2,19 @@
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
 using System;
+using System.Collections.Concurrent;
```

Comment:
JsonResult? You might want to convert all of this new logic to a shared file.

Reply:
yup yup. Totally forgot about Json